### PR TITLE
Allow for passing specification directly, skipping the config file

### PR DESCRIPTION
**Pull request by unindented**
Pull request to get the ball rolling. Change whatever you feel necessary.

### CODE REVIEW

File: lib/appdmg.js [hunk 1]
```diff
@@ -10,24 +10,38 @@ var util = require('./util');
...
+  var source = config.source;
+  var target = config.target;
+
+  if (!basepath) {
+    if (source) {
+      basepath = path.dirname(source);
+    } else {
+      basepath = process.cwd();
+    }
+  }
```
Comment (unindented) [inline lib/appdmg.js]: If `basepath` is not set, it uses the `source` dirname, or `process.cwd()` if we are passing `specification` directly.

File: lib/appdmg.js [hunk 2]
```diff
@@ -10,24 +10,38 @@ var util = require('./util');
...
+    return path.resolve(basepath, to);
   };
 
   var global = {
-    source: source,
     target: target
   };
 
-  delete source;
-  delete target;
```
Comment (unindented) [inline lib/appdmg.js]: Don't delete variables: http://jslinterrors.com/only-properties-should-be-deleted

**Comment by unindented** [comment]
@LinusU if you have time, would you mind taking a look at this PR? It'd allow other projects (e.g. https://github.com/rakuten-frontend/grunt-appdmg/pull/1) to avoid having to create temp files.

**Comment by LinusU** [comment]
Did this a bit different, 1e631c35dc1c719e3738dd1a9b32dd608b7c953c

**Comment by LinusU** [comment]
Sorry for taking so long and thanks for nagging me :)

**Comment by unindented** [comment]
@LinusU Awesome, thanks!

**Pull request closed without merging**